spec/controllers/ingredient_recipe_controller_spec.rb

```diff
@@ -3,78 +3,62 @@
 RSpec.describe IngredientRecipeController, type: :controller do
   render_views
 
-  describe "GET #new" do
 
+  let!(:recipe) { create(:recipe) }
+  let!(:ingredient) { create(:ingredient,:with_name) }
+  let!(:ingredient_recipe) { create(:ingredient_recipe, :valid)}
+
+
+  describe "GET #new" do
     it "adds" do
-      r = Recipe.create!
-      ing = Ingredient.create!(:name => "Ing")
-      post :new, format: :json,:recipe_id => r.id,:ingredient_id => ing.id,:quantity => 1,:unit => "g"
-       expect(response).to have_http_status(:success)
+      post :new, format: :json,:recipe_id => recipe.id,:ingredient_id => ingredient.id,:quantity => 1,:unit => "g"
+      expect(response).to have_http_status(:success)
       expect(IngredientRecipe.find_by_quantity(1)).to_not be_nil
     end
 
     it "returns error when recipe missing" do
-      ing = Ingredient.create!(:name => "Ing")
-      post :new, format: :json,:recipe_id => nil,:ingredient_id => ing.id,:quantity => 1,:unit => "g"
-       expect(response).to have_http_status(500)
+      post :new, format: :json,:recipe_id => nil,:ingredient_id => ingredient.id,:quantity => 1,:unit => "g"
+      expect(response).to have_http_status(500)
```

Comment:
be more precise in your test , 500 is server error , 
you can use status code 422 in this case